### ISSUE TITLE
How to export string values?

### ISSUE THREAD

**Issue by RufusJWB** [lapo-luchini/asn1js#75]
I often have the situation that I have long strings embedded in an ASN.1 data structure:

![image](https://github.com/lapo-luchini/asn1js/assets/1633499/97c51cbd-0032-4e4e-b571-75e70cdd1d40)

I'd like to be able to extract this string from the UI. Is this possible?

**Comment by RufusJWB** [comment]
Any updates on this?

**Comment by olibu** [comment]
I once added a function to copy the hex value of the selected node. Would it help to provide a context menu to copy the node as hex or as string?

Could look like this:
![image](https://github.com/lapo-luchini/asn1js/assets/7242556/8d6a97e3-4077-4a8e-b01a-4c0fc0efc7d3)

**Comment by RufusJWB** [comment]
I would love both options. Probably I'd prefer "Copy String"

**Comment by olibu** [comment]
You can try my feature branch https://github.com/olibu/asn1js/tree/feature/hexcopy
I do not have an ASN1 with large Strings inside to test it with Strings.

Just click on the hex value you want to extract.

**Comment by RufusJWB** [comment]
> You can try my feature branch https://github.com/olibu/asn1js/tree/feature/hexcopy

Do you have it deployed somewhere online?

**Comment by olibu** [comment]
no. You can download the branch as zip, extract it locally and open the
index.html
Maybe I find some time to deploy it somewhere, but this will take some time.

**Comment by RufusJWB** [comment]
> no. You can download the branch as zip, extract it locally and open the index.html Maybe I find some time to deploy it somewhere, but this will take some time.

I got it working. Thank you!

"Copy as Hex" is working for me but "Copy as String" is not. And I don't find it very intuitive that I need to click on the hex dump. I'd rather like to click on the node in the tree. Would this also be possible?

**Comment by olibu** [comment]
There is already a click handler on the node view. Thus I decided to add
the function to the hex view.
Moving the function to the node view would lead into a lot of rework.
I would spend some time on it. But only if there is a chance to merge it to
the main repository. It‘s quite hard for me to keep this up to date.

**Comment by RufusJWB** [comment]
> There is already a click handler on the node view. Thus I decided to add the function to the hex view. Moving the function to the node view would lead into a lot of rework.

--> In that case, I don't think it is necessary to invest a lot of time. But I would appreciate if also the "Copy as String" would be working

**Comment by olibu** [comment]
Update is available.
I added a "Copy as Pretty", too.
Do you have an example of a "complex String"?

**Comment by olibu** [comment]
I fixed an issue with the position of the context menu after scrolling. And the current content is exported to a string if it is no set or sequence.
If you don't have any further point I'm going to make a pull request for this.

**Comment by lapo-luchini** [comment]
I merged that into [branch github-82](https://github.com/lapo-luchini/asn1js/tree/github-82), not ready to inclusion in trunk yet, but you can use it right now [on the branch via githack](https://rawcdn.githack.com/lapo-luchini/asn1js/github-82/index.html).

**Comment by lapo-luchini** [comment]
I forgot to write it here, this this has [been online](https://asn1js.eu/) for a couple of weeks.
@RufusJWB is that what you wanted, or not yet?
